### PR TITLE
DOP-2140: Add site title and version dropdown to sidenav

### PR DESCRIPTION
**Pull request by rayangler**
### Stories/Links:

DOP-2140

### Staging Links:

[BI-Connector](https://docs-mongodbcom-staging.corp.mongodb.com/master/bi-connector/raymundrodriguez/DOP-2140/) (with version dropdown)
[Drivers](https://docs-mongodbcom-staging.corp.mongodb.com/master/drivers/raymundrodriguez/DOP-2140/) (without dropdown)
[Landing](https://docs-mongodborg-staging.corp.mongodb.com/upstream-nav/landing/raymundrodriguez/DOP-2140/)

### Notes:
* Added the site's `title` as a prop and the `VersionDropdown` to our `Sidenav` component.
* Changed the default value of `setBannerContent` in `header-context` because of a complaining test.
* Changed the use of `PropTypes.oneOf` to `PropTypes.oneOfType` in `RootProvider` to make prop check valid.
* Added `ProductsList` as part of the IA transition.

### CODE REVIEW

File: src/components/VersionDropdown.js [hunk 1]
```diff
@@ -96,6 +97,7 @@ const VersionDropdown = ({
       placeholder={null}
       size={Size.Large}
       value={parserBranch}
+      usePortal={false}
```
[the remark this answers is not in the document]
Comment (sophstad) [reply src/components/VersionDropdown.js:100]: Ahhh got it *shakes fist*

Thanks for explaining!

File: src/components/VersionDropdown.js [hunk 2]
```diff
@@ -16,7 +17,7 @@ const zip = (a, b) => {
 };
 
 const StyledSelect = styled(Select)`
```
Comment (sophstad) [inline src/components/VersionDropdown.js:19]: `Select` appears to be the same shade of grey as the sidebar -- I think its background is `transparent`. Perhaps we should make it white?

File: src/layouts/index.js [hunk 3]
```diff
@@ -79,10 +79,18 @@ const DefaultLayout = (props) => {
     <>
       <Global styles={globalCSS} />
       <SiteMetadata siteTitle={title} />
-      <RootProvider isSidebarEnabled={isSidebarEnabled} pageTitle={pageTitle} selectors={page?.options?.selectors}>
```
Comment (sophstad) [inline src/layouts/index.js:82]: If we're no longer passing `pageTitle` as a prop, should it be removed from `RootProvider` and `NavigationContext` altogether?

Comment (rayangler) [reply src/layouts/index.js:82]: Thanks for catching this!

**Review by sophstad — APPROVED** [review]

**Pull request merged**